### PR TITLE
Order Fulfillment Auto > Packaged

### LINKED ISSUE #52: User Story 31, All Merchants fulfill items on an order
```
When all items in an order have been "fulfilled" by their merchants
The order status changes from "pending" to "packaged"
```

DO THESE STORIES FIRST:
- User Story 53, Merchant fulfills part of an order

### Checklist:

- [x] base tests are written
- [x] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #274

depends on #210

### PR DESCRIPTION
**Pull request by BrennanAyers**
This PR brings in the functionality of an Order automatically changing from Pending to Packaged if all of its OrderItems have been fulfilled. This requires a #check_fulfillments method on the Order to .all? the OrderItems for their fulfillment status.
#check_fulfillments gets called by each OrderItem after_save and when :fulfilled is changed. This causes the Order to #check_fulfillments every time an OrderItem:fulfilled is updated.

Resolves #52

### CODE REVIEW

File: app/models/order.rb [hunk 1]
```diff
@@ -24,4 +24,10 @@ def grand_total
       item.price * item.quantity
     end
   end
+
+  def check_fulfillments
+    if order_items.all? {|order_item| order_item.fulfilled}
+      update(status: :packaged)
```
Comment (zachlp2016) [inline app/models/order.rb:30]: Nice!

File: app/models/order_item.rb [hunk 2]
```diff
@@ -5,7 +5,7 @@ class OrderItem < ApplicationRecord
   validates_presence_of :quantity, :price
   validates_inclusion_of :fulfilled, in: [true, false]
 
-  after_save :update_inventory
+  after_save :update_inventory, :check_order
```
Comment (zachlp2016) [inline app/models/order_item.rb:8]: Excellent

**Review by zachlp2016 — APPROVED** [review]
Looks great!
